### ISSUE TITLE
Hue Motion Sensor light level - dark=false and daylight=false?

### ISSUE THREAD

**Issue by driskell** [ebaauw/homebridge-hue#83]
Hello,

Not sure whether this is an issue in Hue or something that maybe should be handled better on the Homebridge side. At a specific time during the day we are seeing dark=false and daylight=false. Of course, we're just going to go ahead and set up automation triggers using just one of those settings, and adjust the sensitivity as necessary. Though it was a little confusing to start as we used Dark=Yes and Daylight=Yes as the two different triggers for whether it is daylight or not. Of course, when both false, it didn't make sense, we'd be seeing "Not daylight" inside Hue app but no automation trigger.

I wonder what your thoughts are on this? Is it worth making the homebridge-hue only ever report them mutually exclusive - that is, never both the same value? Might be less confusing as it does seem to be it might be some internal calculation thing inside Hue and that it would make more sense when exposing it to Humans for it to be mutually exclusive.

Thanks! Great work and really appreciate it.

**Comment by ebaauw** [comment]
> Not sure whether this is an issue in Hue

The `Dark` and `Daylight` characteristics that homebridge-hue exposes simply take the values of the `dark` and `daylight` attributes on the Hue bridge.  Philips did a brilliant job of not documenting their intended use.
> I wonder what your thoughts are on this? 

Here's what I figured out:
- The `config` attributes `tholddark` and `tholdoffset` provide the thresholds for `dark` and `daylight`:   
`dark = (lightlevel < tholddark)`  
`daylight = (lightlevel > tholddark + tholdoffset)`
- Last time I looked, the official Philips Hue creates rules that only look at `dark`.  This causes the light to switch on and off continuously, when the Hue motion sensor "looks" straight at a light, causing `dark` to be false when the light is on;
- HomeKit rules execute when any one of the triggers fires while all conditions are true;
- Unlike HomeKit, the Hue bridge doesn't recognise the difference between triggers and conditions.  Hue bridge rules execute when a (the "last") condition _becomes_ true, while the other conditions are true already.

> Though it was a little confusing to start as we used Dark=Yes and Daylight=Yes as the two different triggers for whether it is daylight or not.

A HomeKit rule with these triggers would fire when `Dark` becomes true _or_ when `Daylight` becomes true, _not_ when (`Dark` and `Daylight`) becomes true.  With a positive value for `tholdoffset`, `Dark` and `Daylight` can never be true at the same time, so a Hue bridge rule with these conditions would never fire.

> Is it worth making the homebridge-hue only ever report them mutually exclusive - that is, never both the same value? 

There's three possible situations:
1. `Dark`, which implies not `Daylight`;
2. `Daylight`, which implies not `Dark`;
3. "Twilight", not `Dark` and not `Daylight`.

I use 1) for rules to turn on the lights, and 2) for rules to turn off the lights.  During twilight, I don't change the lights.  With the right calibration of `tholddark` and `tholdoffset` the lights turn on automatically when I close the blinds and off when I open them.

I have set `tholddark` to `12000` for all my rooms, and `tholdoffset` between `2000` and `8000`, depending on the `lightlevel` when the lights are on (and there's no natural daylight).  Note that `lightlevel` gets updated only once every five minutes or so, when no motion is detected, so double-check `lastupdated` before noting down the `lightlevel`.
